src/test/java/cc/fasttext/DictionaryTest.java

```diff
@@ -155,4 +158,23 @@ public static String readWord(java.io.Reader reader) throws IOException {
         return sb.length() == 0 ? null : sb.toString();
     }
 
+    @Test
+    public void testComputeSubwords() {
+        ImmutableMap<String, Set<String>> wordToSubwords = ImmutableMap.of(
```

Comment:
The test cases are generated by https://ideone.com/XR19v4 .

Reply:
well, it looks like both versions (c++ and yours java) match.